packagehandlers/conanpackagehandler.go

```diff
@@ -0,0 +1,94 @@
+package packagehandlers
+
+import (
+	"errors"
+	"fmt"
+	"github.com/jfrog/frogbot/v2/utils"
+	"github.com/jfrog/jfrog-client-go/utils/io/fileutils"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	conanTechName = "conan"
+	conanFileTxt  = "conanfile.txt"
+	conanFilePy   = "conanfile.py"
+)
+
+type ConanPackageHandler struct {
+	CommonPackageHandler
+}
+
+func (conan *ConanPackageHandler) UpdateDependency(vulnDetails *utils.VulnerabilityDetails) error {
+	if vulnDetails.IsDirectDependency {
+		return conan.updateDirectDependency(vulnDetails)
+	} else {
+		return &utils.ErrUnsupportedFix{
+			PackageName:  vulnDetails.ImpactedDependencyName,
+			FixedVersion: vulnDetails.SuggestedFixedVersion,
+			ErrorType:    utils.IndirectDependencyFixNotSupported,
+		}
+	}
+}
+
+func (conan *ConanPackageHandler) updateDirectDependency(vulnDetails *utils.VulnerabilityDetails) (err error) {
+	isConanFileTxtExists, err := fileutils.IsFileExists(conanFileTxt, false)
+	if err != nil {
+		err = fmt.Errorf("failed while serching for conanfile in project: %s", err.Error())
+		return
+	}
+	if isConanFileTxtExists {
+		if err = conan.updateConanFile(conanFileTxt, vulnDetails); err != nil {
+			return
+		}
+		return conan.installConanDependencies(conanFileTxt)
+	}
+	isConanFilePyExists, err := fileutils.IsFileExists(conanFilePy, false)
+	if err != nil {
+		err = fmt.Errorf("failed while serching for conanfile in project: %s", err.Error())
+		return
+	}
+	if isConanFilePyExists {
+		if err = conan.updateConanFile(conanFilePy, vulnDetails); err != nil {
+			return
+		}
+		return conan.installConanDependencies(conanFilePy)
+	}
+	// If no conanfile found, return an error
+	return fmt.Errorf("failed to update conan dependency: conanfile not found")
+
+}
+
+func (conan *ConanPackageHandler) updateConanFile(conanFileName string, vulnDetails *utils.VulnerabilityDetails) (err error) {
+	var fixedFile string
+	wd, err := os.Getwd()
+	if err != nil {
+		return
+	}
+	filePath := filepath.Clean(filepath.Join(wd, conanFileName))
```

Comment:
will the descriptor always in this path? we have the detection of the descriptors location but we will not necessarily be in the WD where the descriptor is. For example what about multi-module scenario? should we support it?

Reply:
added the logic of locating all the descriptors in the project, and added a test case to test a more complex scenario